File: src/backend/cmd/server/brand_environment.go
package main

import "os"

const legacyEnvironmentPrefix = "AGENTHUB_" // [brand-compat] 仅兼容读取旧部署变量。

func readDiAgentEnv(suffix string) string {
	value, _ := lookupDiAgentEnv(suffix)
	return value
}

func lookupDiAgentEnv(suffix string) (string, bool) {
	if value, ok := os.LookupEnv("DI_AGENT_" + suffix); ok {
		return value, true
	}
	return os.LookupEnv(legacyEnvironmentPrefix + suffix)
}

File: src/frontend/electron/brand-environment.mjs
const LEGACY_PREFIX = 'AGENTHUB_'; // [brand-compat] 旧桌面环境变量只读入口。
const CANONICAL_PREFIX = 'DI_AGENT_';

export function readDiAgentEnvironment(env, suffix) {
  const canonicalKey = `${CANONICAL_PREFIX}${suffix}`;
  if (Object.hasOwn(env, canonicalKey)) {
    return env[canonicalKey];
  }
  return env[`${LEGACY_PREFIX}${suffix}`];
}

/** 子进程只接收 canonical 名称，避免继续传播旧环境变量。 */
export function canonicalizeDiAgentEnvironment(env) {
  const canonical = {};
  for (const [key, value] of Object.entries(env)) {
    if (key.startsWith(LEGACY_PREFIX)) {
      const canonicalKey = `${CANONICAL_PREFIX}${key.slice(LEGACY_PREFIX.length)}`;
      if (!Object.hasOwn(env, canonicalKey)) {
        canonical[canonicalKey] = value;
      }
      continue;
    }
    canonical[key] = value;
  }
  return canonical;
}
